src/expandable-section/interfaces.ts

```diff
@@ -45,12 +46,33 @@ export interface ExpandableSectionProps extends BaseComponentProps {
   children?: React.ReactNode;
 
   /**
-   * Heading displayed above the content text.
-   *
-   * When using the container variant, use the Header component. Otherwise, use plain text.
+   * **Deprecated**, use headerText instead.
+   * @deprecated
    */
   header?: React.ReactNode;
 
+  /**
+   * Heading displayed above the content text. When using the container variant, use it with additional header props. Otherwise, use plain text.
+   */
+  headerText?: React.ReactNode;
+
+  /**
+   * Supplementary text below the heading. Use with container variant.
+   */
+  headerDescription?: string;
+
+  /**
+   * Specifies secondary text that's displayed to the right of the heading title. Use with container variant.
+   * Behaves similar to the Header component counter.
+   */
+  headerCounter?: string;
+
+  /**
+   * Overrides the default [HTML heading tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements)
+   * h2. Use with container variant.
+   */
+  headingTagOverride?: ExpandableSectionProps.HeadingTag;
```

Comment:
Did we decide to keep this prop? I remember a discussion that the expected tag for expandable sections is always h2

Reply:
We removed `headingTagVariant`, always using h2 tag and style. `headingTagOverride` can overwrite the tag but not style.